### DISCUSSION TITLE
Data Plane HTTP extension: Need example

### DISCUSSION THREAD

**Discussion by logicalfarhad** [eclipse-edc/Connector#1857] [Q&A]
Hi there, 
I want to add different rest endpoints as assets in EDC. Then after successful contract negotiation, I want to call the external rest endpoint with some parameters with the asset id, 
which will return the JSON result. I also want to pass some query parameters from the transfer request process.  In summary, the consumer will get some JSON response from a specific rest endpoint, and the provider will provide the JSON result.

Any suggestion or idea on how I can achieve this?

Thanks in advance.

**Comment by ndr-brt** [comment]
Hi, I'm not sure I understood your needs, but the `data-plane-http` extension should be what you're looking for.
Ref: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/extensions/data-plane/data-plane-http

**Comment by hlange** [comment]
I wrote something that could help: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/discussions/2107#discussioncomment-4224961
